core/spec/models/spree/product_spec.rb

```diff
@@ -547,17 +549,30 @@ class Extension < Spree::Base
     end
 
     it 'should return sum of stock items count_on_hand' do
-      product = create(:product)
       product.stock_items.first.set_count_on_hand 5
       product.variants_including_master.reload # force load association
       expect(product.total_on_hand).to eql(5)
     end
 
     it 'should return sum of stock items count_on_hand when variants_including_master is not loaded' do
-      product = create(:product)
       product.stock_items.first.set_count_on_hand 5
       expect(product.reload.total_on_hand).to eql(5)
     end
+
+    context 'when the stock items are loaded' do
+      it 'returns the loaded count_on_hand instead of doing SUM(count_on_hand)' do
+        product.stock_items.first.set_count_on_hand(5)
+
+        product = described_class.includes(:stock_items).find(product.id)
```

Comment:
Not sure this override works properly since it is failing on CircleCI. Also spec setup should be in a `before` block.